### PR TITLE
[sparsehash] Add new port

### PR DESCRIPTION
**Pull request by zi-m**
https://github.com/sparsehash/sparsehash

### CODE REVIEW

File: ports/sparsehash/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+else()
+
+	vcpkg_execute_required_process(
+		COMMAND ${SOURCE_PATH}/configure ${OPTIONS} --prefix=${CURRENT_PACKAGES_DIR}
+		WORKING_DIRECTORY ${SOURCE_PATH}
+		LOGNAME configure-${TARGET_TRIPLET}
+	)
+
+	vcpkg_execute_required_process(
+		COMMAND make -j install
```
Comment (cbezault) [inline ports/sparsehash/portfile.cmake]: Why run `make install` with the `-j` option? Either way don't ever call `make` with a bare `-j`. It probably doesn't do what you want. I suggest replacing that with `make -j ${VCPKG_CONCURRENCY}`.

File: ports/sparsehash/00001-windows-use-std.patch [hunk 2]
```diff
@@ -0,0 +1,17 @@
+diff --git a/src/windows/sparsehash/internal/sparseconfig.h b/src/windows/sparsehash/internal/sparseconfig.h
+index 3091559..f96e08c 100644
+--- a/src/windows/sparsehash/internal/sparseconfig.h
++++ b/src/windows/sparsehash/internal/sparseconfig.h
+@@ -7,10 +7,10 @@
+ #define GOOGLE_NAMESPACE  ::google
+ 
+ /* the location of the header defining hash functions */
+-#define HASH_FUN_H  <hash_map>
```
Comment (cbezault) [inline ports/sparsehash/00001-windows-use-std.patch]: This would be worth upstreaming, hash_map is super old.

**Review by cbezault — CHANGES_REQUESTED** [review]

File: ports/sparsehash/portfile.cmake (hunk 1 above)
Comment (zi-m) [reply ports/sparsehash/portfile.cmake]: Using the patch I replace `<hash_map>` (line 9) with `<unordered_map>` (line 10).
On windows there is no `<hash_map>`.

File: ports/sparsehash/00001-windows-use-std.patch (hunk 2 above)
Comment (zi-m) [reply ports/sparsehash/00001-windows-use-std.patch]: Using the patch I replace `<hash_map>` (line 9) with `<unordered_map>` (line 10).
On windows there is no `<hash_map>`.

File: ports/sparsehash/portfile.cmake (hunk 1 above)
Comment (zi-m) [reply ports/sparsehash/portfile.cmake]: Fixed.

File: ports/sparsehash/00001-windows-use-std.patch (hunk 2 above)
Comment (cbezault) [reply ports/sparsehash/00001-windows-use-std.patch]: It's just super old https://docs.microsoft.com/en-us/cpp/standard-library/hash-map-class?view=vs-2019. Many libraries provided their own hash_map implementations before c++11.

**Pull request merged**